server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -376,11 +377,28 @@ void dispatchRequest(final RestRequest restRequest, final RestChannel channel, f
             if (badRequestCause != null) {
                 dispatcher.dispatchBadRequest(channel, threadContext, badRequestCause);
             } else {
+                populateRequestThreadContext0(restRequest, channel, threadContext);
                 dispatcher.dispatchRequest(restRequest, channel, threadContext);
```

Comment:
This is the code that clears and scopes the thread context for a particular http request to be handled.
This is now extended to also populate the said thread context (with selected HTTP header values).

Reply:
Is this happening before or after authentication ?

Reply:
In this PR this is happening **after** authentication (on `main` it happens before authentication).

The `populateRequestThreadContext` hook method is used to instante the authentication context, for the authn that happened earlier.